src/main/java/codeu/controller/ActivityFeedServlet.java

```diff
@@ -1,16 +1,203 @@
+    super.init();
+    setConversationStore(ConversationStore.getInstance());
+    setMessageStore(MessageStore.getInstance());
+    setUserStore(UserStore.getInstance());
+  }
+
+  /**
+   * Sets the UserStore used by this servlet. This function provides a common setup method for use
+   * by the test framework or the servlet's init() function.
+   */
+  void setUserStore(UserStore userStore) {
+    this.userStore = userStore;
+  }
+
+  /**
+   * Sets the ConversationStore used by this servlet. This function provides a common setup method
+   * for use by the test framework or the servlet's init() function.
+   */
+  void setConversationStore(ConversationStore conversationStore) {
+    this.conversationStore = conversationStore;
+  }
+
+  /**
+   * Sets the MessageStore used by this servlet. This function provides a common setup method for use
+   * by the test framework or the servlet's init() function.
+   */
+  void setMessageStore(MessageStore messageStore) {
+    this.messageStore = messageStore;
+  }
+
+  /**
+   * Formats creation time into String.
+   */
+  private String formatTime(Instant creationTime) {
+    DateTimeFormatter formatter =
+        DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")
+            .withZone( ZoneId.systemDefault() );
+    //ZoneId.of("America/Chicago")
+    return formatter.format(creationTime);
+  }
+  /**
+   * Formats User data into readable string.
+   */
+  private Activity formatUser(User user) {
+    String creationTime = formatTime(user.getCreationTime());
+    String output = user.getName() + " joined Charmer Chat.";
+    return new Activity(creationTime,output);
+  }
+  /**
+   * Formats Conversation data into readable string.
+   */
+  private Activity formatConversation(Conversation conversation) {
+    User owner = userStore.getUserWithId(conversation.getOwnerId());
+    String creationTime = formatTime(conversation.getCreationTime());
+    String output = owner.getName() + " started conversation " + conversation.getTitle();
+    return new Activity(creationTime,output);
+  }
+  /**
+   * Formats Message data into readable string.
+   */
+  private Activity formatMessage(Message message) {
+    User author = userStore.getUserWithId(message.getAuthorId());
+    Conversation conversation = conversationStore.getConversationWithId(message.getConversationId());
+    String creationTime = formatTime(message.getCreationTime());
+    System.out.println(author);
+    System.out.println(conversation);
+    System.out.println(message);
+    String output = author.getName() + " sent in " + conversation.getTitle() + ": " + message.getContent();
+    return new Activity(creationTime,output);
+  }
+
+  /**
+   * Gets all activity on app, iterates through to sort by timestamp and prepares printed statements.
+   */
+  List<Activity> getActivities() {
+    List<Activity> activities = new LinkedList<>();
+    LinkedList<Conversation> conversations = new LinkedList<>(conversationStore.getAllConversations());
```

Comment:
Do we assume that these are sorted in some way? I don't see anything in the docstrings for the data stores that say they guarantee a returned order.

We might want to either guarantee that as part of the API (and add a test for it), or put sorts in place here

Reply:
They are now being sorted in the `GetActivities` function.